src/components/Autocomplete.tsx

```diff
@@ -0,0 +1,91 @@
+import React, {
+  ChangeEvent, useMemo, useState,
+} from 'react';
+import debounce from 'lodash.debounce';
+import 'bulma';
+import { peopleFromServer } from '../data/people';
+
+interface AutocompleteProps {
+  onSelected: (person: string) => void;
+}
+
+export const Autocomplete: React.FC<AutocompleteProps> = ({ onSelected }) => {
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const [showSuggestions, setShowSuggestions] = useState(false);
+
+  const applyQuery = React.useMemo(
+    () => debounce(setAppliedQuery, 1000),
+    [],
+  );
+
+  const applyList = React.useMemo(
+    () => debounce(setShowSuggestions, 1100),
+    [],
+  );
+
+  const handleQueryChange = (event: ChangeEvent<HTMLInputElement>) => {
+    const newQuery = event.target.value;
+
+    setQuery(newQuery);
+    applyQuery(newQuery);
+    applyList(!!newQuery);
+  };
+
+  const filterPeople = useMemo(() => {
+    return peopleFromServer.filter(user => (
+      user.name.toLowerCase().includes(appliedQuery.toLowerCase())
+    ));
+  }, [appliedQuery]);
+
+  const handleSuggestion = (person: string) => {
+    setQuery(person);
+    setShowSuggestions(false);
+    onSelected(person);
+  };
+
+  const handleBlur = () => {
+    setTimeout(() => {
+      setShowSuggestions(false);
+    }, 200);
```

Comment:
```suggestion
      setShowSuggestions(false);
```
if instead onClick set onMouseDown event - no need will be use setTimeout here)

Reply:
Got it, thanks for the tip) I'll try to remember